lib/active_job/queue_adapters/sidekiq_publisher_adapter.rb

```diff
@@ -11,19 +11,28 @@ class SidekiqPublisherAdapter
       JOB_WRAPPER_CLASS = ActiveJob::QueueAdapters::SidekiqAdapter::JobWrapper.to_s.freeze
 
       def enqueue(job)
-        internal_enqueue(job)
+        if ActiveRecord::Base.connection.transaction_open?
```

Comment:
🎱 Think it's worth creating an internal method to use that checks for the transaction being open instead of repeating it? Thought being we don't want to know much about the internals here and it might be helpful if we ever have to change how we detect it for whatever reason.

Reply:
+1 to that idea, though not a blocker

Reply:
[25735d8](https://github.com/ezcater/sidekiq_publisher/pull/74/commits/25735d8d1fcfd917ee1dabb51c8ab4a7239f331a)